send-transaction-service/src/create_client_for_tests.rs

```diff
@@ -0,0 +1,75 @@
+//! This module contains functionality required to create tests parametrized
+//! with the client type.
+
+use {
+    crate::{
+        tpu_info::NullTpuInfo,
+        transaction_client::{
+            spawn_tpu_client_send_txs, Cancelable, ConnectionCacheClient, TpuClientNextClient,
+            TransactionClient,
+        },
+    },
+    solana_client::connection_cache::ConnectionCache,
+    std::{net::SocketAddr, sync::Arc},
+    tokio::runtime::Handle,
+};
+
+// `maybe_runtime` argument is introduced to be able to use runtime from test
+// for the TpuClientNext, while ConnectionCache uses runtime created internally
+// in the quic-client module and it is impossible to pass test runtime there.
+pub trait CreateClient: TransactionClient {
+    fn create_client(
+        maybe_runtime: Option<Handle>,
```

Comment:
This might be a questionable solution. But the good news is that as soon as we deprecate `ConnectionCache`, this code will be simplified. Alternatively, I could have changed the interface of the quic-client to take runtime instead of using the runtime defined internally, but I would prefer to not touch it.